src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -312,6 +312,14 @@ public static bool IsUnnecessaryCast(this CastExpressionSyntax cast, SemanticMod
             var expressionTypeInfo = semanticModel.GetTypeInfo(cast.Expression, cancellationToken);
             var expressionType = expressionTypeInfo.Type;
 
+            if (expressionType != null &&
+                expressionType.IsEnumType() 
+                && cast?.Parent is PrefixUnaryExpressionSyntax prefixUnaryExpression 
+                && prefixUnaryExpression.Kind() == SyntaxKind.UnaryMinusE xpression)
```

Comment:
:bulb: The type check is slow, but `IsKind()` is fast. Also, we know from code above that `cast` cannot be null, and the `IsKind` extension method handles null internally (so `cast.Parent` is covered). You can just use this for the parent check:

```
cast.Parent.IsKind(SyntaxKind.UnaryMinusExpression)
```